See http://sources.redhat.com/ml/libc-hacker/2005-02/msg00000.html
and http://www.mail-archive.com/pld-cvs-commit@pld-linux.org/msg00229.html

Fixes
  In file included from jis0208.c:23:
  jis0208.h:32: error: array type has incomplete element type
when building glibc with gcc-4.0
The bug has been present since at least glibc-2.2.5.
This patch applies cleanly to glibc-2.3.4

--- /home/dank/downloads/glibc-2.3-20050307/iconvdata/jis0208.h	2003-06-11 14:40:42.000000000 -0700
+++ glibc-2.3-20050307/iconvdata/jis0208.h	2005-03-13 20:55:01.784054760 -0800
@@ -24,15 +24,6 @@
 #include <gconv.h>
 #include <stdint.h>
 
-/* Conversion table.  */
-extern const uint16_t __jis0208_to_ucs[];
-
-extern const char __jisx0208_from_ucs4_lat1[256][2];
-extern const char __jisx0208_from_ucs4_greek[0xc1][2];
-extern const struct jisx0208_ucs_idx __jisx0208_from_ucs_idx[];
-extern const char __jisx0208_from_ucs_tab[][2];
-
-
 /* Struct for table with indeces in UCS mapping table.  */
 struct jisx0208_ucs_idx
 {
@@ -42,6 +33,15 @@
 };
 
 
+/* Conversion table.  */
+extern const uint16_t __jis0208_to_ucs[];
+
+extern const char __jisx0208_from_ucs4_lat1[256][2];
+extern const char __jisx0208_from_ucs4_greek[0xc1][2];
+extern const struct jisx0208_ucs_idx __jisx0208_from_ucs_idx[];
+extern const char __jisx0208_from_ucs_tab[][2];
+
+
 static inline uint32_t
 __attribute ((always_inline))
 jisx0208_to_ucs4 (const unsigned char **s, size_t avail, unsigned char offset)
